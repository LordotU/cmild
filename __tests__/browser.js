/* eslint-disable no-undef */

process.browser = true

import mockConsole from 'jest-mock-console'

import TestClassDecoratedWithoutParams from './mocks/TestClassDecoratedWithoutParams'
import TestClassDecoratedWithParams from './mocks/TestClassDecoratedWithParams'


describe('Cmild for Browser', () => {

  describe('Without params', () => {

    let outputs
    let instance

    beforeEach(() => {
      outputs = []

      mockConsole({
        debug: (...args) => outputs.push([...args]),
      })

      instance = new TestClassDecoratedWithoutParams()
    })

    test('Logs non-async methods', async () => {

      const testArgValue = 'testArgValue'
      expect(instance.testMethod1(testArgValue)).toBe(testArgValue)

      const erroredArgValue = 'erroredArgValue'
      expect(() => instance.erroredMethod1(erroredArgValue)).toThrowError(erroredArgValue)

      expect(outputs).toMatchSnapshot()
    })

    test('Logs async methods', async () => {

      const testArgValue1 = 'testArgValue1'
      const testArgValue2 = 'testArgValue2'
      await expect(instance.testMethod2(testArgValue1, testArgValue2)).resolves.toBe(`${testArgValue1} ${testArgValue2}`)

      const erroredArgValue1 = 'erroredArgValue1'
      const erroredArgValue2= 'erroredArgValue2'
      await expect(instance.erroredMethod2(erroredArgValue1, erroredArgValue2)).rejects.toThrowError(`${erroredArgValue1} ${erroredArgValue2}`)

      expect(outputs).toMatchSnapshot()
    })
  })

  describe('With params', () => {

    let outputs
    let instance

    beforeEach(() => {
      outputs = []

      mockConsole({
        debug: (...args) => outputs.push([...args]),
      })

      instance = new TestClassDecoratedWithParams()
    })

    test('Logs non-async methods', async () => {

      const testArgValue = 'testArgValue'
      expect(instance.testMethod1(testArgValue)).toBe(testArgValue)

      const erroredArgValue = 'erroredArgValue'
      expect(() => instance.erroredMethod1(erroredArgValue)).toThrowError(erroredArgValue)

      expect(outputs).toMatchSnapshot()
    })

    test('Logs async methods', async () => {

      const testArgValue1 = 'testArgValue1'
      const testArgValue2 = 'testArgValue2'
      await expect(instance.testMethod2(testArgValue1, testArgValue2)).resolves.toBe(`${testArgValue1} ${testArgValue2}`)

      const erroredArgValue1 = 'erroredArgValue1'
      const erroredArgValue2= 'erroredArgValue2'
      await expect(instance.erroredMethod2(erroredArgValue1, erroredArgValue2)).rejects.toThrowError(`${erroredArgValue1} ${erroredArgValue2}`)

      expect(outputs).toMatchSnapshot()
    })
  })

})
