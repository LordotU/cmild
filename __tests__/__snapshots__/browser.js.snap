// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cmild for Browser With params Logs async methods 1`] = `
Array [
  Array [
    "%c TestClass#testMethod2 given arguments: %O",
    "background: #03A9F4; color: #FFFFFF; font-weight: italic",
    Array [
      "testArgValue1",
      "testArgValue2",
    ],
  ],
  Array [
    "%c TestClass#testMethod2 result: %O",
    "background: #4CAF50; color: #FFFFFF; font-weight: italic",
    "testArgValue1 testArgValue2",
  ],
  Array [
    "%c TestClass#erroredMethod2 given arguments: %O",
    "background: #03A9F4; color: #FFFFFF; font-weight: italic",
    Array [
      "erroredArgValue1",
      "erroredArgValue2",
    ],
  ],
  Array [
    "%c TestClass#testMethod2 given arguments: %O",
    "background: #03A9F4; color: #FFFFFF; font-weight: italic",
    Array [
      "erroredArgValue1",
      "erroredArgValue2",
    ],
  ],
  Array [
    "%c TestClass#testMethod2 result: %O",
    "background: #4CAF50; color: #FFFFFF; font-weight: italic",
    "erroredArgValue1 erroredArgValue2",
  ],
  Array [
    "%c TestClass#erroredMethod2 error: %O",
    "background: #F20404; color: #FFFFFF; font-weight: italic",
    [Error: erroredArgValue1 erroredArgValue2],
  ],
]
`;

exports[`Cmild for Browser With params Logs non-async methods 1`] = `
Array [
  Array [
    "%c TestClass#testMethod1 given arguments: %O",
    "background: #03A9F4; color: #FFFFFF; font-weight: italic",
    Array [
      "testArgValue",
    ],
  ],
  Array [
    "%c TestClass#testMethod1 result: %O",
    "background: #4CAF50; color: #FFFFFF; font-weight: italic",
    "testArgValue",
  ],
  Array [
    "%c TestClass#erroredMethod1 given arguments: %O",
    "background: #03A9F4; color: #FFFFFF; font-weight: italic",
    Array [
      "erroredArgValue",
    ],
  ],
  Array [
    "%c TestClass#testMethod1 given arguments: %O",
    "background: #03A9F4; color: #FFFFFF; font-weight: italic",
    Array [
      "erroredArgValue",
    ],
  ],
  Array [
    "%c TestClass#testMethod1 result: %O",
    "background: #4CAF50; color: #FFFFFF; font-weight: italic",
    "erroredArgValue",
  ],
  Array [
    "%c TestClass#erroredMethod1 error: %O",
    "background: #F20404; color: #FFFFFF; font-weight: italic",
    [Error: erroredArgValue],
  ],
]
`;

exports[`Cmild for Browser Without params Logs async methods 1`] = `
Array [
  Array [
    "%c TestClass#testMethod2 given arguments: %O",
    "background: #F2EAFF; color: #03A9F4; font-weight: bold",
    Array [
      "testArgValue1",
      "testArgValue2",
    ],
  ],
  Array [
    "%c TestClass#testMethod2 result: %O",
    "background: #F2EAFF; color: #4CAF50; font-weight: bold",
    "testArgValue1 testArgValue2",
  ],
  Array [
    "%c TestClass#erroredMethod2 given arguments: %O",
    "background: #F2EAFF; color: #03A9F4; font-weight: bold",
    Array [
      "erroredArgValue1",
      "erroredArgValue2",
    ],
  ],
  Array [
    "%c TestClass#testMethod2 given arguments: %O",
    "background: #F2EAFF; color: #03A9F4; font-weight: bold",
    Array [
      "erroredArgValue1",
      "erroredArgValue2",
    ],
  ],
  Array [
    "%c TestClass#testMethod2 result: %O",
    "background: #F2EAFF; color: #4CAF50; font-weight: bold",
    "erroredArgValue1 erroredArgValue2",
  ],
  Array [
    "%c TestClass#erroredMethod2 error: %O",
    "background: #F2EAFF; color: #F20404; font-weight: bold",
    [Error: erroredArgValue1 erroredArgValue2],
  ],
]
`;

exports[`Cmild for Browser Without params Logs non-async methods 1`] = `
Array [
  Array [
    "%c TestClass#testMethod1 given arguments: %O",
    "background: #F2EAFF; color: #03A9F4; font-weight: bold",
    Array [
      "testArgValue",
    ],
  ],
  Array [
    "%c TestClass#testMethod1 result: %O",
    "background: #F2EAFF; color: #4CAF50; font-weight: bold",
    "testArgValue",
  ],
  Array [
    "%c TestClass#erroredMethod1 given arguments: %O",
    "background: #F2EAFF; color: #03A9F4; font-weight: bold",
    Array [
      "erroredArgValue",
    ],
  ],
  Array [
    "%c TestClass#testMethod1 given arguments: %O",
    "background: #F2EAFF; color: #03A9F4; font-weight: bold",
    Array [
      "erroredArgValue",
    ],
  ],
  Array [
    "%c TestClass#testMethod1 result: %O",
    "background: #F2EAFF; color: #4CAF50; font-weight: bold",
    "erroredArgValue",
  ],
  Array [
    "%c TestClass#erroredMethod1 error: %O",
    "background: #F2EAFF; color: #F20404; font-weight: bold",
    [Error: erroredArgValue],
  ],
]
`;
